File: packages/rune-webserver/ios/RuneWebServer.swift
//
//  RuneWebServer.swift
//  RuneWebServer
//
//  Public interface for RuneWebServer module
//

import Foundation
import RuneKit

@objc(RuneWebServer)
public class RuneWebServer: NSObject {
  private static var moduleInstance: RuneWebServerModule?

  @objc public static func initialize(with runtime: RuneRuntime) {
    guard moduleInstance == nil else {
      print("[RuneWebServer] Module already initialized")
      return
    }

    let module = RuneWebServerModule()
    runtime.installModules([module])
    moduleInstance = module
    print("[RuneWebServer] Module initialized")
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}

File: packages/rune-webserver/android/RuneWebServer/src/main/java/dev/rune/webserver/RuneWebServer.kt
package dev.rune.webserver

import android.app.Activity
import android.util.Log
import com.rune.kit.runtime.RuneRuntime

/**
 * Public interface for RuneWebServer module.
 */
object RuneWebServer {
    private var moduleInstance: RuneWebServerModule? = null

    @JvmStatic
    fun initialize(activity: Activity, runtime: RuneRuntime) {
        if (moduleInstance != null) {
            Log.w("RuneWebServer", "Module already initialized")
            return
        }

        moduleInstance = RuneWebServerModule(activity)
        runtime.installModules(listOf(moduleInstance!!))
        Log.d("RuneWebServer", "Module initialized")
    }

    @JvmStatic
    fun cleanup() {
        moduleInstance?.shutdown()
        moduleInstance = null
    }
}
